legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/ServiceStoreExecutionExtension.java

```diff
@@ -42,4 +51,28 @@ public List<Function3<ExecutionNode, MutableList<CommonProfile>, ExecutionState,
             return null;
         }));
     }
+
+    public List<Function5<SecurityScheme, Credential, HttpConnectionBuilder, Identity, CredentialProviderProvider, Boolean>> getExtraCredentialProcessors()
+    {
+        return Collections.singletonList((securityScheme, credential, httpConnectionBuilder, identity, credentialProviderProvider) ->
+        {
+            if (securityScheme instanceof HttpSecurityScheme)
+            {
+                String cred = credential.accept(new ServiceStoreCredentialConsumer());
```

Comment:
how is this approach extensible for new consumers?

Reply:
We can create new consumers internally for other type of credentials. But there should be only one way how we use a specific type of credential in context of a specific store. Otherwise, we again need to decide which way to choose. Plus, i cant think of a scenario where there can be multiple ways to use a credential for a specific store.